bin/epaa.py

```diff
@@ -1206,24 +1211,26 @@ def __main__():
         for k in first_entry.keys():
             complete_df['{} log2 protein LFQ intensity'.format(k)] = complete_df.apply(
                 lambda row: create_quant_column_value_for_result(row, protein_quant, transcriptSwissProtMap, k), axis=1)
-    # parse (differential) expression analysis results, annotate features (genes/transcripts)
-    if args.gene_expression is not None:
-        fold_changes = read_diff_expression_values(args.gene_expression)
-        gene_id_lengths = {}
-        col_name = 'RNA expression (RPKM)'
-
-        with open(args.gene_reference, 'r') as gene_list:
-            for l in gene_list:
-                ids = l.split('\t')
-                gene_id_in_df = complete_df.iloc[1]['gene']
-                if 'ENSG' in gene_id_in_df:
-                    gene_id_lengths[ids[0]] = float(ids[2].strip())
-                else:
-                    gene_id_lengths[ids[1]] = float(ids[2].strip())
-        deseq = False
-        # add column to result dataframe
-        complete_df[col_name] = complete_df.apply(lambda row: create_expression_column_value_for_result(
-            row, fold_changes, deseq, gene_id_lengths), axis=1)
+    # parse expression (nf-core/rnaseq) analysis results, annotate features (genes/transcripts)
+    if args.expression is not None:
+        rnaseq_results = pd.read_csv(args.expression, sep='\t', header=0)
+
+        measure = "count" if "count" in args.expression else "TPM"
```

Comment:
Is the name of the measure type always in the rnaseq filename? Might be also an option to check the colnames within the file since we read in the file anyways

Reply:
According to the `rnaseq` pipeline output I saw it is always contained in the filename. You are right, that it is not the most robust solution but I think you can't tell from the column names which type it is.

Reply:
Alright, then lets keep it like that :)